### PR TITLE
Remove TransactionAttributes

### PR DESCRIPTION
**Pull request by erikzhang**

### CODE REVIEW

File: src/neo/Network/P2P/Payloads/TransactionVersion.cs [hunk 1]
```diff
@@ -0,0 +1,12 @@
+using System;
+
+namespace Neo.Network.P2P.Payloads
+{
+    [Flags]
+    public enum TransactionVersion : byte
+    {
+        Default = 0,
+
+        All = 0
```
Comment (shargon) [inline src/neo/Network/P2P/Payloads/TransactionVersion.cs]:
```suggestion
        All = Default
```

Comment (erikzhang) [reply src/neo/Network/P2P/Payloads/TransactionVersion.cs]: It should be updated when new element is added.

**Comment by erikzhang** [comment]
Do you want me to remove `TransactionVersion` first?

**Comment by shargon** [comment]
Yes, I prefer in two PR, version could be changed to enum for both cases.

**Comment by erikzhang** [comment]
Done.

**Comment by nicolegys** [comment]
Test passed.

Works well, "attributes" is removed and the tx size is decreased by 1byte.
send the same tx,
**when `attributes` is not removed**:
size is 265 and `"attributes"` exists
image

**when `attributes` is removed**:
size is 264 and `"attributes"` dosen't exist
image

@Tommo-L We need to make corresponding adjustments on neo-cli and neo-modules after this pr.

**Comment by roman-khimov** [comment]
> when attributes is not removed:
> size is 265 and "attributes" exists
> ...
> when attributes is removed:
> size is 264 and "attributes" dosen't exist

So it's 0,4% size improvement, is it really worth it? What is the problem we're to trying to solve here? I think attributes is a nice extensible metadata mechanism that can be used for many purposes.

**Comment by superboyiii** [comment]
@erikzhang Since we need to make corresponding adjustments on neo-cli and neo-modules after this pr, please don't merge this to `Preview2` until it released.

**Comment by erikzhang** [comment]
@superboyiii We use daily build, so it doesn't matter whether I merge or not.

@neo-project/core We leave a byte `0` in the attributes position, which can provide a possibility for adding new fields in the future. What do you think?

**Comment by shargon** [comment]
@neo-project/core We leave a byte 0 in the attributes position, which can provide a possibility for adding new fields in the future. What do you think?

Then what's the difference? I think that we should solve first how will enter first the OracleTx and then the UserTx, and I think that the best way is with `TransactionAttributes`

**Comment by erikzhang** [comment]
> I think that we should solve first how will enter first the OracleTx and then the UserTx, and I think that the best way is with TransactionAttributes

The best way is with version flags. We should move `ExpectedHash` to the parameter of the contract method. So there will be no need to keep the attributes.

**Comment by shargon** [comment]
But the memory pool need to know the UserTx's hash related to this OracleTx, so it should be parsed without processing the script. 
Or you prefer something like that?

```csharp
class Transaction
{
public UInt256 UserTx{get;} = null;

OnDeserialize(Stream s)
{
     if (Version==OracleTx) UserTx=s.Read<Uint256>();
     else UserTx=null;
}
}

```

**Comment by erikzhang** [comment]
OK. I'll close it then.

**Pull request closed without merging**